File: app/src/main/java/net/xanir/kariyerassignment/restApi/responses/MainResponse.java
package net.xanir.kariyerassignment.restApi.responses;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by Umur Kaya on 18-Sep-19.
 */
public class MainResponse {

    @SerializedName("date")
    @Expose
    private String date;
    @SerializedName("month")
    @Expose
    private String month;
    @SerializedName("marketName")
    @Expose
    private String marketName;
    @SerializedName("orderName")
    @Expose
    private String orderName;
    @SerializedName("productPrice")
    @Expose
    private Float productPrice;
    @SerializedName("productState")
    @Expose
    private String productState;
    @SerializedName("productDetail")
    @Expose
    private ProductDetail productDetail;

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getMonth() {
        return month;
    }

    public void setMonth(String month) {
        this.month = month;
    }

    public String getMarketName() {
        return marketName;
    }

    public void setMarketName(String marketName) {
        this.marketName = marketName;
    }

    public String getOrderName() {
        return orderName;
    }

    public void setOrderName(String orderName) {
        this.orderName = orderName;
    }

    public Float getProductPrice() {
        return productPrice;
    }

    public void setProductPrice(Float productPrice) {
        this.productPrice = productPrice;
    }

    public String getProductState() {
        return productState;
    }

    public void setProductState(String productState) {
        this.productState = productState;
    }

    public ProductDetail getProductDetail() {
        return productDetail;
    }

    public void setProductDetail(ProductDetail productDetail) {
        this.productDetail = productDetail;
    }
}

File: app/src/main/java/net/xanir/kariyerassignment/restApi/responses/MainResponse.kt
package net.xanir.kariyerassignment.restApi.responses

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by Umur Kaya on 18-Sep-19.
 */
class MainResponse {

    @SerializedName("date")
    @Expose
    var date: String? = null
    @SerializedName("month")
    @Expose
    var month: String? = null
    @SerializedName("marketName")
    @Expose
    var marketName: String? = null
    @SerializedName("orderName")
    @Expose
    var orderName: String? = null
    @SerializedName("productPrice")
    @Expose
    var productPrice: Float? = null
    @SerializedName("productState")
    @Expose
    var productState: String? = null
    @SerializedName("productDetail")
    @Expose
    var productDetail: ProductDetail? = null
}
